tracee-rules/signatures/rego/regosig/traceerego.go

```diff
@@ -183,24 +190,52 @@ func (sig *RegoSignature) OnEvent(e types.Event) error {
 	}
 
 	if len(results) > 0 && len(results[0].Expressions) > 0 && results[0].Expressions[0].Value != nil {
-		switch v := results[0].Expressions[0].Value.(type) {
-		case bool:
-			if v {
-				sig.cb(types.Finding{
-					Data:        nil,
-					Context:     ee,
-					SigMetadata: sig.metadata,
-				})
-			}
+
+		switch results[0].Expressions[0].Value.(type) {
 		case map[string]interface{}:
+			values, ok := results[0].Expressions[0].Value.(map[string]interface{})
+			if ok && len(values) <= 0 {
+				return nil
+			}
+		case bool:
+			sig.dispatch(results[0].Expressions[0].Value, ee)
+		}
+
+		// For AIO: result set can be an empty set of length=1, so we need to check value
```

Comment:
I think our goal to have new implementation of types.Signature for AIO was to really don't touch existing RegoSignature.OnEvent method. Otherwise it's not different from adding aioEnabled arg to the NewRegoSignature constructor